go/private/rules/stdlib.bzl

```diff
@@ -39,12 +40,13 @@ def _stdlib_impl(ctx):
   goroot = root_file.path[:-(len(root_file.basename)+1)]
   sdk = ""
   for f in ctx.files._host_sdk:
-    prefix, found, extension  = f.path.partition("bin/go")
+    prefix, found, _  = f.path.partition("bin/go")
     if found:
       sdk = prefix
+      break
   if not sdk:
     fail("Could not find go executable in go_sdk")
-  go = ctx.actions.declare_file("bin/go" + extension)
+  go = ctx.actions.declare_file("bin/go" + goos_to_extension(ctx.attr.goos))
```

Comment:
Is this right? Seems like this would add .exe if you were cross-compiling for Windows from Linux.

Reply:
Nope, it's not right, realized as soon as I uploaded it.
The old code was broken though because it failed to actually hold on to the extension, and when I fixed that it was sometimes finding other binaries that start with the same string (gofmt).
I am finding a different solution...